### PR TITLE
feat(linter): fix nullish coalescing

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**
## Description

replace || with ?? to all files
Also, added disable linters in two places in this PR as we need OR condition and by using ?? the logic was not working

## Why

To remove lint errors

## Issue

n/a

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/shared/basic/Input/BasicInput.tsx [hunk 1]
```diff
@@ -136,18 +136,21 @@ const BasicInput = ({
...
-                onClick={toggleHidden}
-              />
-            ) : (
-              <VisibilityOffIcon
-                style={{ marginLeft: '8px', color: Colors.secondary }}
-                onClick={toggleHidden}
-              />
-            ))}
+          {
+            // eslint-disable-next-line
```
Comment (manojava-gk) [inline src/components/shared/basic/Input/BasicInput.tsx:140]: Kindly provide an explanation for this disable linter

File: src/components/shared/basic/Input/BasicInput.tsx [hunk 2]
```diff
@@ -157,14 +160,17 @@ const BasicInput = ({
...
-        {(errorMessage || (hint && focus)) && (
-          <Typography
-            sx={{ color: errorMessage ? Colors.error : Colors.secondary }}
-            variant={'label4'}
-          >
-            {errorMessage ?? hint}
-          </Typography>
-        )}
+        {
+          // eslint-disable-next-line
```
Comment (manojava-gk) [inline src/components/shared/basic/Input/BasicInput.tsx:164]: Kindly provide an explanation for this disable linter

**Review by manojava-gk — COMMENTED** [review]
Kindly document the reason for disable linter

File: src/components/shared/basic/Input/BasicInput.tsx (hunk 1 above)
Comment (nidhigarg-bmw) [reply src/components/shared/basic/Input/BasicInput.tsx:140]: Added disable linters here as we need OR condition here and by using ?? the logic was not working

File: src/components/shared/basic/Input/BasicInput.tsx (hunk 2 above)
Comment (nidhigarg-bmw) [reply src/components/shared/basic/Input/BasicInput.tsx:164]: Added disable linters here as we need OR condition and by using ?? the logic was not working

Comment (jjeroch) [reply src/components/shared/basic/Input/BasicInput.tsx:164]: is there no better solution?

Comment (nidhigarg-bmw) [reply src/components/shared/basic/Input/BasicInput.tsx:164]: @jjeroch Nullish operator (??) return RHS only if LHS returns null or undefined but here first condition is boolean which either return true/false. Cases which is falling into false or empty, we have to disable it.

**Review by oyo — APPROVED** [review]

**Review by oyo — COMMENTED** [review]
please resolve conflicts

**Review by oyo — APPROVED** [review]

**Pull request merged**